src/og/entity/GeoObjectHandler.js

```diff
@@ -0,0 +1,1106 @@
+import * as shaders from "../shaders/geoObject.js";
+import { concatArrays, makeArrayTyped, spliceArray } from "../utils/shared.js";
+
+const VERTEX_BUFFER = 0,
+    POSITION_BUFFER = 1,
+    RGBA_BUFFER = 2,
+    NORMALS_BUFFER = 3,
+    INDECIES_BUFFER = 4,
+    DIRECTION_BUFFER = 5,
+    PITCH_ROLL_BUFFER = 6,
+    SIZE_BUFFER = 7,
+    PICKINGCOLOR_BUFFER = 8;
+
+const setParametersToArray = (arr = [], index = 0, length, itemSize, ...params) => {
+    const currIndex = index * length;
+    for (let i = currIndex; i < currIndex + length; i++) {
+        arr[i] = params[i % itemSize];
+    }
+    return arr;
+};
+
+class GeoObjectHandler {
+    constructor(entityCollection) {
+        /**
+         * Picking rendering option.
+         * @public
+         * @type {boolean}
+         */
+        this.pickingEnabled = true;
+
+        this._entityCollection = entityCollection;
+        this._renderer = null;
+        this._planet = null;
+
+        this._geoObjects = [];
+        this._pitchRollArr = [[]];
+        this._sizeArr = [[]];
+        this._vertexArr = [[]];
+        this._positionHighArr = [[]];
+        this._positionLowArr = [[]];
+        this._directionArr = [[]];
+        this._rgbaArr = [[]];
+        this._normalsArr = [[]];
+        this._indicesArr = [[]];
+        this._pickingColorArr = [[]];
+
+        this._pitchRollBuffer = [];
+        this._sizeBuffer = [];
+        this._vertexBuffer = [];
+        this._positionHighBuffer = [];
+        this._positionLowBuffer = [];
+        this._directionBuffer = [];
+        this._rgbaBuffer = [];
+        this._normalsBuffer = [];
+        this._indicesBuffer = [];
+        this._pickingColorBuffer = [];
+
+        this.__staticId = GeoObjectHandler._staticCounter++;
+        this._buffersUpdateCallbacks = [];
+        this._buffersUpdateCallbacks[PICKINGCOLOR_BUFFER] = this.createPickingColorBuffer;
+        this._buffersUpdateCallbacks[POSITION_BUFFER] = this.createPositionBuffer;
+        this._buffersUpdateCallbacks[DIRECTION_BUFFER] = this.createDirectionBuffer;
+        this._buffersUpdateCallbacks[NORMALS_BUFFER] = this.createNormalsBuffer;
+        this._buffersUpdateCallbacks[RGBA_BUFFER] = this.createRgbaBuffer;
+        this._buffersUpdateCallbacks[INDECIES_BUFFER] = this.createIndicesBuffer;
+        this._buffersUpdateCallbacks[VERTEX_BUFFER] = this.createVertexBuffer;
+        this._buffersUpdateCallbacks[SIZE_BUFFER] = this.createSizeBuffer;
+        this._buffersUpdateCallbacks[PITCH_ROLL_BUFFER] = this.createPitchRollBuffer;
+
+        this._changedBuffers = new Array(this._buffersUpdateCallbacks.length);
+
+        this._instancedTags = new Map();
+        this._instancedTags.set("none", {
+            index: 0
+        });
+    }
+
+    getObjectByIndex(index) {
+        return this._geoObjects[index];
+    }
+
+    //Create buffers
+    createVertexBuffer() {
+        var h = this._renderer.handler;
+        for (let i = 0; i < this._vertexArr.length; i++) {
```

Comment:
I prefer `for (let i = 0, len = this._vertexArr.length; i < len; i++)` because length of the array size doesn't change.